### ISSUE TITLE
Refactor to Move parseError to @mogami/solana

### ISSUE THREAD

**Issue by RamirezAlex** [kin-labs/kinetic#259]
The logic of this [function](https://github.com/kin-labs/mogami/blob/857fae964c959cc06399480fcfdb93d75e5a4bbf/libs/api/app/data-access/src/lib/helpers/parse-error.ts) should live in the backend, that way the backend throws appropriate error types and we won't need to reimplement a parser function on every SDK.

We are using `sendRawTransaction` which uses [sendEncodedTransaction](https://github.com/solana-labs/solana/blob/da28badcaaec75b25fc308a47741ea6da500eaeb/web3.js/src/connection.ts#L4221) which doesn't return the error type. Fine a better way to parse it.

```js
{
  jsonrpc: '2.0',
  error: {
    code: -32002,
    message: 'Transaction simulation failed: Error processing Instruction 1: custom program error: 0x1',
    data: {
      accounts: null,
      err: { InstructionError: [ 1, { Custom: 1 } ] },
      logs: [
        'Program Memo1UhkJRfHyvLMcVucJwxXeuD728EqVDDwQDxFMNo invoke [1]',
        'Program Memo1UhkJRfHyvLMcVucJwxXeuD728EqVDDwQDxFMNo consumed 374 of 1400000 compute units',
        'Program Memo1UhkJRfHyvLMcVucJwxXeuD728EqVDDwQDxFMNo success',
        'Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA invoke [1]',
        'Program log: Instruction: Transfer',
        'Program log: Error: insufficient funds',
        'Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA consumed 2559 of 1399626 compute units',
        'Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA failed: custom program error: 0x1'
      ],
      unitsConsumed: 374
    }
  },
  id: '8079ddfb-db2f-4707-b7d0-870b5f8cc7b0'
}
```

**Comment by RamirezAlex** [comment]
Include this TODOs to make we return proper error messages
https://github.com/kin-labs/kinetic/pull/266/files#diff-93f22001e668ed9c5b2d902e0eb36936db2aa77b6ef0339593a8a0674f9f594bR57
https://github.com/kin-labs/kinetic/pull/266/files#diff-7c2e0ccadbcfc9bbfaaeb9e85aa8938a750d4f72c19f09fb0fee29f9b085fc8eR89
https://github.com/kin-labs/kinetic/pull/266/files#diff-7c2e0ccadbcfc9bbfaaeb9e85aa8938a750d4f72c19f09fb0fee29f9b085fc8eR102

**Comment by RamirezAlex** [comment]
Closed by: #287
